src/Avalonia.Controls/Primitives/PopupPositioning/IPopupPositioner.cs

```diff
@@ -452,16 +453,14 @@ static class PopupPositionerExtensions
             PopupPositionerConstraintAdjustment constraintAdjustment, Rect? rect,
             FlowDirection flowDirection)
         {
-            // We need a better way for tracking the last pointer position
-#pragma warning disable CS0618 // Type or member is obsolete
-            var pointer = topLevel.PointToClient(topLevel.PlatformImpl?.MouseDevice.Position ?? default);
-#pragma warning restore CS0618 // Type or member is obsolete
-
             positionerParameters.Offset = offset;
             positionerParameters.ConstraintAdjustment = constraintAdjustment;
             if (placement == PlacementMode.Pointer)
             {
-                positionerParameters.AnchorRectangle = new Rect(pointer, new Size(1, 1));
+                // We need a better way for tracking the last pointer position
+                var position = topLevel.PointToClient(topLevel.LastPointerPosition ?? default);
```

Comment:
I couldn't find a better idea, except keeping old property but internal.
I don't think we can really subscribe on popup's target' root pointer movements, as there might not be any movements between assigning the target and opening the popup.